src/nimib.nim

```diff
@@ -74,6 +81,8 @@ template nbInit*() =
     #   - in case you need to manage additional exceptions for a specific document add a new set of partials before calling nbSave
     nbDoc.context.searchDirs(nbDoc.templateDirs)
     nbDoc.context.searchTable(nbDoc.partials)
+    when defined(nimibSrcDir):
+      nbDoc.filename = (nbDoc.filename.toAbsoluteDir.relativeTo nimibSrcDirAbs).string
```

Comment:
this says `toAbsoluteDir` but it should actually be a `toAbsoluteFile`. Should we use [toAbsolute](https://nim-lang.org/docs/compiler/pathutils.html#toAbsolute%2Cstring%2CAbsoluteDir) with nbHomeDir as base?

Reply:
Oops, good catch should indeed be File and not Dir 😄 
Not really sure exactly how you intend to use `toAbsolute` here. Add it to the existing code or replace the existing code with it? It has a check that if the file-path is already absolute it will ignore the the base.

Reply:
Not sure either how to use it but there is not a `toAbsoluteFile` available right?

Reply:
Oh right, just assumed it did exist... Yes then we have to use it or `AbsoluteFile(file)` which it uses internally. That is if we know that `nbDoc.filename` is a absolute path to begin with...

If `nbDoc.filename` is absolute (currently the case) it doesn't matter what `base` is because it won't be used. If it on the other hand is a relative file then we'd want it to return its actual path in the `nimibSrcDir` to be able to give the correct relative path later for saving the file in `nbHomeDir`. This is why we need `nimibSrcDir` to begin with. So it depends on what the relative path is relative to I guess. But I don't think using `nbHomeDir` as `base` will work in any of the cases.

Reply:
Ok think I've got this sorted now. We split it into two scenarios:
- `nbDoc.filename.isAbsolute`: We calculate the relative path from `nimibSrcDir`
- `isRelative`: we do nothing, we trust the user has already set it to the appropriate relative path. 

How does this sound?

Reply:
sounds good! 🚀